### PR TITLE
Improve localization of urls in jetpack cloud

### PR DESCRIPTION
**Pull request by robertsreberski**
Related to D106031-code

## Proposed Changes

After migrating the jetpack.com URL structure from locale subdomains to subdirectories - we investigate Jetpack Cloud to ensure links follow the updated structure. This PR makes changes to ensure links are correct in the most visible spots.

## Testing Instructions

* Checkout this PR
* Go to `http://jetpack.cloud.localhost:3001/es/pricing`
* Click around and make sure the general structure of links to `jetpack.com` pages follow `https://jetpack.com/<locale>`
* Go to `https://jetpack.cloud.localhost:3001/es/features/comparison`
* Click around and make sure that (product page) URLs follow new structure; whilst support links are still linking to EN (unfortunately these have many 404s for localized versions)

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by dlind1 — APPROVED** [review]
LGTM!

File: client/jetpack-cloud/sections/comparison/table/links.ts [hunk 1]
```diff
@@ -21,11 +23,11 @@ export const links = {
 	payments_block: 'https://jetpack.com/support/jetpack-blocks/payments-block/',
 	priority_support: 'https://jetpack.com/features/security/expert-priority-support/',
 	related_posts: 'https://jetpack.com/features/traffic/related-posts/',
-	scan: 'https://jetpack.com/upgrade/scan/',
-	search: 'https://jetpack.com/upgrade/search/',
+	scan: localizeUrl( 'https://jetpack.com/upgrade/scan/' ),
+	search: localizeUrl( 'https://jetpack.com/upgrade/search/' ),
 	secure_authentication: 'https://jetpack.com/features/security/secure-authentication/',
```
Comment (jboland88) [inline client/jetpack-cloud/sections/comparison/table/links.ts:28]: Just curious why we are localizing some but not all of these links for jetpack.com? Are some of the other pages not reliably translated?

**Review by jboland88 — APPROVED** [review]
This LGTM - I noticed that there are some URLs on the feature comparison page that we are not localizing here - just curious why that is?

File: client/jetpack-cloud/sections/comparison/table/links.ts (hunk 1 above)
Comment (robertsreberski) [reply client/jetpack-cloud/sections/comparison/table/links.ts:28]: Yes, I've decided to localize product URLs which are translated into all languages. Feature or support pages are unfortunately not consistent in this.

**Pull request merged**